queries/scala/indents.scm

```diff
@@ -0,0 +1,32 @@
+[
+  (template_body)
+  (block)
+  (parameters)
+  (arguments)
+  (match_expression)
+  (splice_expression)
+  (import_declaration)
+  (function_definition)
```

Comment:
I think function definitions should only be indented like this if it doesn't have a `(block)` within it, otherwise the block will get doubly indented. Maybe replace this with a
```query
(function_definition
  body: (_) @_body
  (#not-kind-eq? @_body "block")) @indent.begin
```
below? Or only apply `(block)` indents when not in a `function_definition`? Whatever you think makes the most sense given the grammar.

Reply:
For some reason it doesn't cause any problems. Consider
```scala
def foo() = {
  println() // Indented only once
```

So, I guess we could leave it like this 🤔

Reply:
what about 
```scala
def foo() = 
  {
    println()
}
```

Reply:
Scala has scalafmt which is a standard formatter that is included in the language server and is used almost in every Scala project.
So, I perceive indentation queries only as a tool to guess non-irritating and grammatically correct indentation while typing.

Reply:
> For some reason function_definitions with blocks are not double indented. Consider
> 
> ```scala
> def foo() = {
>   println() // Indented only once
> }
> ```
> 
> So, I guess we could leave it like this, with simpler queries 🤔

Interesting... looks like it only applies when using multiline parameters. See the following:

```scala
  def foo(
    sourceDir: File
  ) = {
      println() // Indented twice
    }
```